src/EntityRepository/CustomerRepository.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\EntityRepository;
+
+use App\Entity\Customer;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+
+/**
+ * @ORM\Entity
+ */
+class CustomerRepository extends ServiceEntityRepository
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function getEntityClass(): string
+    {
+        return Customer::class;
```

Comment:
galbūt norima eiti kitu keliu (sprendžiant iš ServiceEntityRepository - autowiring), tačiau mano nuomone tinkamesnis kelias norint gauti entity repozitoriją būtų išskirtinai ją paimti iš pačio Entity su 'getRepository' funkcija, kuri yra EntityManagerInterface. Tuomet šis kodo gabaliukas mums tampa nereikalingas, o taip pat turime laisviau skaitoma kodą (pvz.:

private function getCustomerRepository(): CustomerRepository|EntityRepository
{
    return $this-getEntityManager()->getRepository(Customer::class);
}

$result = $this-getCustomerRepository()->findOneBy(['code' => $example]);